### ISSUE TITLE
EML.XML validation not supported or broken?

### ISSUE THREAD

**Issue by CecSve** [gbif/gbif-data-validator#98]
A GBIF publisher is experiencing issues with validation of his eml.xml file in the data validator.

Data validation result [here](https://www.gbif.org/tools/data-validator/3c57ac02-17d4-48bd-82c6-52c2a273994c):

Regardless, it was able to parse out three issues with the .eml file

**1.  "The licence can not be parsed, is not supported by GBIF or is simply missing”**

- I’ve attached the relevant EML file, which includes a license section as follows:
 <eml:license>
      <eml:licenseName>CC-BY 4.0</eml:licenseName>
      <eml:url>[https://creativecommons.org/licenses/by/4.0/</eml:url>](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcreativecommons.org%2Flicenses%2Fby%2F4.0%2F%253C%2Feml%3Aurl%253E&data=05%7C01%7Calbenson%40usgs.gov%7C26bedaae94e447e2648808da23d495c3%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637861693272823997%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=Lfy2yWcSJNE1rT3sedFKIg0Aj4BBDmr0QG2SLqo%2BTEo%3D&reserved=0)
    </eml:license>
Is there something wrong with that?

**2.  I also get "The description of the dataset is missing or too short” - which I don’t understand.**

and lastly, for the EML file:

**3.  The EML document does not validate against the schema”**

ERROR
cvc-complex-type.3.2.2: Attribute 'core_scope' is not allowed to appear in element 'eml:eml'.
cvc-complex-type.4: Attribute 'scope' must appear on element 'eml:eml'.
cvc-complex-type.2.4.a: Invalid content was found starting with element '{"eml://ecoinformatics.org/eml-2.1.1":dataset}'. One of '{dataset}' is expected.

Not sure what any of that means. I built this EML file from our old EML file that worked and a few other places.

**Comment by CecSve** [comment]
The first issue could be due to changes in EML versions where _license_ has been changed to _licensed_.

Perhaps the validation will work as DWC with only EML, but not as a xml file?

**Comment by MattBlissett** [comment]
What's the EML version? We support version 1.0, 1.0.1, 1.0.2 and 1.1 at present, and have a longer term task to update to the newer versions.

(The version is probably in the first few lines of the EML).

**Comment by CecSve** [comment]
eml-2.1.1

Then I suppose the first issue is not fixed by changing license to licensed either which was introduced with version 2.

**Comment by AbbyBenson** [comment]
@tmcelrath I'm trying to figure out the best way to proceed on this. I could help with building an EML file using the IPT that you could work with as your starting point instead? Would that be helpful?

Also note that when I tried to load your eml.xml file into XMLNotepad it gave this error:
<img width="216" alt="XMLNotepadError" src="https://user-images.githubusercontent.com/7586150/167162077-6632e982-9cf6-4427-9ca2-fb784006fdd7.PNG">

**Comment by tmcelrath** [comment]
Idea, can you send me a properly formatted EML file that has been built by the IPT? That way I can compare with what I have. We used the same text on our last upload in 2015 so I don't have anything to compare it to.

**Comment by mjy** [comment]
@tmcelrath I will resolve this: https://github.com/SpeciesFileGroup/taxonworks/issues/2986, and try to validate from there.

**Comment by mjy** [comment]
> Idea, can you send me a properly formatted EML file that has been built by the IPT? 

^ Please attach the very latest version of the EML that is supported to this issue if possible.

**Comment by mjy** [comment]
<strike>Is there an stand-alone web-based EML validator that I can check against for 1.1.2?  I.e. for only that file?</strike> Never mind this.

**Comment by mjy** [comment]
Now I am confused.  Here is EML from a very recent ALA file upload to GBIF.  It appears to use 2.1.  Perhaps our issue is something else.

<img width="2028" alt="Screen Shot 2022-05-17 at 1 08 01 PM" src="https://user-images.githubusercontent.com/22600/168880925-aabe83a6-946e-41f9-a065-5c39736d3dbe.png">

```xml
<?xml version="1.0" encoding="utf-8"?>
<eml:eml xmlns:d="eml://ecoinformatics.org/dataset-2.1.0" xmlns:eml="eml://ecoinformatics.org/eml-2.1.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:dc="http://purl.org/dc/terms/" xsi:schemaLocation="eml://ecoinformatics.org/eml-2.1.1 http://rs.gbif.org/schema/eml-gbif-profile/1.1/eml-gbif-profile.xsd" system="ALA-Registry" scope="system" xml:lang="en">
  <dataset>
  ...
  ```

**Comment by mjy** [comment]
@tmcelrath I believe the problem is the file you supplied is a frankenstein of old and new, for example the section you merged re `licensed` is not valid in the 2.1 version.  I would prefer to have us use the ALA example and update from there.  I'll send you a copy offline.

**Comment by mjy** [comment]
I think I found our main problem:

* Floating around in older EML files, and TW version: [http://rs.gbif.org/schema/eml-gbif-profile/1.1/eml.xsd](http://rs.gbif.org/schema/eml-gbif-profile/1.1/eml.xsd) 
*  ALA version: [http://rs.gbif.org/schema/eml-gbif-profile/1.1/eml-gbif-profile.xsd](http://rs.gbif.org/schema/eml-gbif-profile/1.1/eml-gbif-profile.xsd).

The validator is likely running the second.  We'll update our generator to use that and go from there.

**Comment by mdoering** [comment]
Yes, `eml-gbif-profile.xsd` is the correct schema for the GBIF profile. It uses a subset of the entire EML and adds GBIF specific elements  under the `additionalMetadata` extension slot. The `eml.xsd` is only the part of the EML schema that is used for the GBIF profile. The GBIF additions live in their own namespace and hence need a separate schema file.

**Comment by CecSve** [comment]
@mjy and @tmcelrath were you able to solve this issue?

**Comment by tmcelrath** [comment]
Yes, we were. @mjy what exactly was the solution in the XML stuff at the top of the file?
